### PR TITLE
feat(api): delete and count api routes for product model

### PR DESCRIPTION
**Pull request by Francesco-code1**

### CODE REVIEW

**Review by nullndr — CHANGES_REQUESTED** [review]
🛑 

Please run `npm run format` to properly format the code

File: app/routes/api/v1/warehouse/products/count/by-type/$productTypeId.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  const countedPoducts = await countProductByType(productTypeId);
+
+  if (countedPoducts == null) {
+    throw new Response(null, {
+      status: 404,
+      statusText: "Product Type Not Found",
+    });
+  }
+
+  return countedPoducts;
```
Comment (nullndr) [inline app/routes/api/v1/warehouse/products/count/by-type/$productTypeId.ts:27]: `countedPoducts` should be `countedProducts`.

Also, please do not return directly json values, but return a json object wrapping the value, something like:

```json
{
  numberOfProducts: countedProducts
}
```

In order to correctly return a json response, you could use the [`json`](https://remix.run/docs/en/v1/api/remix#json) utility.

**Review by nullndr — CHANGES_REQUESTED** [review]
🛑

File: app/routes/api/v1/warehouse/products/count/by-status/$productStatus.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    const countedPoductsStatus = await countProductByStatus(parsedData.data);
+
+    if (countedPoductsStatus == null) {
+      throw new Response(null, {
+        status: 404,
+        statusText: "Product Type Not Found",
+      });
+    }
+
+    return JSON.stringify({numberOfProducts: countedPoductsStatus});
```
Comment (nullndr) [inline app/routes/api/v1/warehouse/products/count/by-status/$productStatus.ts:31]: Why this?

The `LoaderData` should be declared as:

type LoaderData = {
  numberOfProducts: number;
};

File: app/routes/api/v1/warehouse/products/count/by-type/$productTypeId.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { LoaderFunction } from "@remix-run/node";
+import { countProductByType } from "~/models/product/product.server";
+
+type LoaderData = string;
```
Comment (nullndr) [inline app/routes/api/v1/warehouse/products/count/by-type/$productTypeId.ts:4]: Again, provide the `LoaderData` as the object, not a string

**Review by nullndr — CHANGES_REQUESTED** [review]
🛑

File: app/routes/api/v1/warehouse/products/count/by-status/$productStatus.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+export const loader: LoaderFunction = async ({
+  params,
+}): Promise<LoaderData> => {
+  const productStatus = params.productStatus;
+
+  if (productStatus == null) {
+    throw new Response(null, {
+      status: 400,
+      statusText: "Product Status Not Provided",
```
[the remark this answers is not in the document]
Comment (Francesco-code1) [reply app/routes/api/v1/warehouse/products/count/by-status/$productStatus.ts:16]: What?

**Review by DaruZero — APPROVED** [review]
🚀

**Comment by DaruZero** [comment]
#51

**Pull request merged**